cdc/sink/mq.go

```diff
@@ -113,13 +113,22 @@ func newMqSink(
 			log.Warn("mqSink treats all updates as inserts if enable old value is not enabled")
 		}
 		var forceHkPk bool
-		forceHKeyToPKey, ok := opts["force-handle-key-pkey"]
-		if ok && forceHKeyToPKey == "true" {
+		forceHkPkOpt, ok := opts["force-handle-key-pkey"]
+		if ok && forceHkPkOpt == "true" {
 			forceHkPk = true
 		}
-		newEncoder1 := newEncoder
+		var supportTxn bool
+		supportTxnOpt, ok := opts["support-txn"]
+		if ok && supportTxnOpt == "true" {
+			supportTxn = true
+		}
 		newEncoder = func() codec.EventBatchEncoder {
-			canalEncoder := newEncoder1().(*codec.CanalEventBatchEncoder)
```

Comment:
Probably need refactoring here, because the original newEncoder would not be used.

Reply:
The latest commit has contained refactor here.